sdk/storage/src/core/errors.rs

```diff
@@ -81,6 +81,12 @@ impl From<azure_core::HttpError> for Error {
     }
 }
 
+impl From<azure_core::StreamError> for Error {
+    fn from(error: azure_core::StreamError) -> Self {
+        Self::CoreError(azure_core::Error::Stream(error))
```

Comment:
I'll make a pass over the storage errors after core & identity ship, but if the goal here is to put all `azure_core::Error` errors `Self::CoreError`, then it can be done with a generic similar to:

https://github.com/Azure/azure-sdk-for-rust/blob/3e928516f9b2d0c9a52bda22427bd7a0e3a57ee6/sdk/data_cosmos/src/errors.rs#L60-L64

but with `Into<azure_core::Error>`.

Reply:
I think this needed to be the variant, since other core variants are handled above in more specific manner. this then leads to duplicate implementations.

Reply:
my bad - this is not even a variant. However i tried your approach and there is conflicting implementations then, and I did not want to change around the error too much in this PR.

Reply:
Sounds good.